### PR TITLE
Woo: Product action-list support for variation deletes

### PR DESCRIPTION
**Pull request by coderkevin**
This adds support to the edit state and action-list for
variation deletes. It omits variations that have a delete edit from
the combined view of edits and API data, and it adds the
deletes to the action list.

*Note for testing: Edit state is not being cleared completely yet, so you'll need to do a browser refresh after you save any changes.*

To test:

1. `npm test`
2. `npm start`
3. Visit `http://calypso.localhost:3000/store/product/<site_id>/<product_id>`
4. Ensure product is variable with several variations (edit and save them if needed, then refresh).
5. Click the "x" next to a variation type option.
6. Observe the variation type option disappears, and the variations involved with that option disappear as well.
7. Click "Update" to save changes via API.
8. Verify that the variation type option no longer exists either by refreshing the page or checking wp-admin interface.

### CODE REVIEW

**Comment by ryelle** [comment]
I ran into some trouble trying to test this. First, I tried creating variations through Calypso, but when I tried adding a second variation, the first disappeared (you can see as soon as a tab to the values field the Color disappears): https://cloudup.com/cQrrbtXTB8v

So I went and created variations in wp-admin, just to test this PR, but when I click the X, I see an error:

screen shot 2017-07-04 at 10 32 19 am

**Comment by coderkevin** [comment]
> when I tried adding a second variation, the first disappeared

@ryelle I think this problem was caused by some code I carried back to previous PRs in this chain, but didn't get all of it pulled back to the same PR. After you tested, I've updated the previous PR (#15779) with the rest of the code it needed. I tested this functionality with the changes and had no further problems.

**Comment by coderkevin** [comment]
> So I went and created variations in wp-admin, just to test this PR, but when I click the X, I see an error

This is a good catch. I was under the incorrect assumption that the id of an attribute would be unique. This is not the case when id = 0, especially, which is unfortunate. I will make the appropriate change, probably upstream in a PR ahead of this one.

File: client/extensions/woocommerce/state/ui/products/edits-reducer.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { isEqual, uniqueId } from 'lodash';
+import { isEqual, isNumber, uniqueId } from 'lodash';
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/ui/products/edits-reducer.js]: `isNumber` doesn't get used, and is causing a linter error.

`'isNumber' is defined but never used`

**Pull request merged**